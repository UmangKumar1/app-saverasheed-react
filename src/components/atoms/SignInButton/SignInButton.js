import React from 'react'
import styled from 'react-emotion'
import PropTypes from 'prop-types'

import {
  authNamePropTypes,
  authIdPropTypes,
} from '../../../fixtures/authProviders'

import Button from '../Button'

const { shape, any } = PropTypes

const IconWrapper = styled('div')({
  // background: 'white',
  padding: 8,
  display: 'flex',
  alignItems: 'center',
  marginRight: 16,
})

const SignInButton = ({ auth: { name, id, Icon }, ...props }) => (
  <Button {...props}>
    <IconWrapper>
      <Icon size={18} />
    </IconWrapper>
    <span>{name}</span>
  </Button>
)

SignInButton.propTypes = {
  auth: shape({
    name: authNamePropTypes.isRequired,
    id: authIdPropTypes.isRequired,
    Icon: any.isRequired,
  }).isRequired,
}

export default SignInButton
