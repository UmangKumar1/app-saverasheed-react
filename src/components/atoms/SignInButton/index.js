export { default, default as SignInButton } from './SignInButton'
