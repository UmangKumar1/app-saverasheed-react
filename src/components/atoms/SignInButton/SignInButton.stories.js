import React from 'react'
import { storiesOf } from '@storybook/react'
import {
  facebookAuth,
  googleAuth,
  passwordAuth,
} from '../../../fixtures/authProviders'
import SignInButton from '.'

storiesOf('SignInButton', module)
  .add('with Google', () => <SignInButton auth={googleAuth} />)
  .add('with Facebook', () => <SignInButton auth={facebookAuth} />)
  .add('with Email', () => <SignInButton auth={passwordAuth} />)
