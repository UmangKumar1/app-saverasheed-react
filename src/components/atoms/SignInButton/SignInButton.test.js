import React from 'react'
import { shallow } from 'enzyme'
import SignInButton from '.'
import { googleAuth } from '../../../fixtures/authProviders'

const wrap = (props = {}) => shallow(<SignInButton {...props} />)

describe('SignInButton', () => {
  it('renders children when passed in', () => {
    const wrapper = wrap({ auth: googleAuth })
    expect(wrapper).toMatchSnapshot()
  })
})
