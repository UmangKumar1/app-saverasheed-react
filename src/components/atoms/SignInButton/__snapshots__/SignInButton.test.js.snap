// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInButton renders children when passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignInButton
    auth={
      Object {
        "Icon": [Function],
        "id": "google.com",
        "method": "signInWithGoogle",
        "name": "Google",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(div)>
          <Google
            size={18}
          />
        </Styled(div)>,
        <span>
          Google
        </span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Google
            size={18}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": 18,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Google",
        },
        "ref": null,
        "rendered": "Google",
        "type": "span",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(div)>
            <Google
              size={18}
            />
          </Styled(div)>,
          <span>
            Google
          </span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Google
              size={18}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "size": 18,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Google",
          },
          "ref": null,
          "rendered": "Google",
          "type": "span",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
